File: entwurfsmuster/java/singleton/Singleton.java
public class Singleton {
    // Private static variable to hold the single instance
    private static Singleton instance;

    // Private constructor to prevent instantiation outside the class
    private Singleton() {
        // Initialization code, if needed
    }

    // Public static method to get the single instance
    public static Singleton getInstance() {
        if (instance == null) {
            instance = new Singleton();
        }
        return instance;
    }

    // Other methods and properties can be added here
    public void someMethod() {
        System.out.println("Some method of the singleton");
    }
}

File: entwurfsmuster/php/singleton.php
class Singleton {
    private static $instance;

    // Private constructor to prevent instantiation outside the class
    private function __construct() {
        // Initialization code, if needed
    }

    // Public static method to get the single instance
    public static function getInstance() {
        if (!isset(self::$instance)) {
            self::$instance = new self();
        }
        return self::$instance;
    }

    // Add other methods as needed
    public function someMethod() {
        echo "Some method of the singleton\n";
    }
}

// Usage
$singleton1 = Singleton::getInstance();
$singleton2 = Singleton::getInstance();

// Check if both instances are the same
var_dump($singleton1 === $singleton2); // Output: bool(true)

// Call a method on the singleton
$singleton1->someMethod(); // Output: Some method of the singleton
$singleton2->someMethod(); // Output: Some method of the singleton
